test/utils.py

```diff
@@ -0,0 +1,46 @@
+import contextlib
+from typing import Any, Optional
+
+import torch
+
+__all__ = ["assert_close"]
+
+try:
+    # torch.testing.assert_close is only available for torch>=1.9
+    from torch.testing import assert_close as _assert_close
+    from torch.testing._core import _get_default_tolerance
+
+    def assert_close(
+        actual: torch.Tensor,
+        expected: torch.Tensor,
+        *,
+        rtol: Optional[float] = None,
+        atol: Optional[float] = None,
+        **kwargs: Any,
+    ) -> None:
+        if rtol is None and atol is None:
+            with contextlib.suppress(Exception):
+                rtol, atol = _get_default_tolerance(actual, expected)
```

Comment:
`torch.testing.assert_close` uses stricter tolerances as `torch.testing.assert_allclose`. There are < 10 tests that fail with the new default tolerances. Thus we use the old ones here.